### PR TITLE
Add Wikiclustering benchmark for multiple languages:))

### PR DESCRIPTION
**Pull request by Rysias**
In the style of [BlurbsClusteringP2P](https://huggingface.co/datasets/slvnwhrl/blurbs-clustering-p2p) and similar datasets, I have created relatively generic benchmarks using top-level Wikipedia categories. I cover the following languages:

- Danish
- Latvian
- Manx
- Albanian

The languages (except for Danish) were arbitrarily chosen as low-resource European languages. 

As evident below, I have not yet run the benchmarks as I cannot access sufficient compute. If I could get help (@KennethEnevoldsen) that'd be 🔥 - otherwise, I will try my best to leave my laptop on overnight:)). 

I look forward to contributing to this great project!

## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [ ] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [ ] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [ ] `intfloat/multilingual-e5-small`
- [ ] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file.

### CODE REVIEW

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        form=None,
+        domains=None,
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        bibtex_citation=None,
+        n_samples={"test": 150_000},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/dan/WikiClusteringP2P.py:31]: downsample to say 2048 pr. language? Should also help you run the models.

Also makes the benchmark faster to run, which we would love.

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+from mteb.abstasks.AbsTaskClustering import AbsTaskClustering
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class WikiClusteringP2P(AbsTaskClustering):
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/dan/WikiClusteringP2P.py:7]: Would make it a multilingual task, you can use the `MultilingualTask` for that. That way you can combine all the tasks into 1

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        description="Clustering of wikipedia articles inspired by BlubrbsClusteringP2P. Labels are taken from top-level categories as defined in https://da.wikipedia.org/wiki/Kategori:Topniveau_for_emner.",
+        reference=None,
+        dataset={
+            "path": "ryzzlestrizzle/dawiki-clustering-p2p",
+            "revision": "4c4c6b6ad63311d6a072734d86dca1c0b48c441c",
+        },
+        type="Clustering",
+        category="p2p",
+        eval_splits=["test"],
+        eval_langs=["dan-Latn"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/dan/WikiClusteringP2P.py:19]:
```suggestion
        eval_langs={"{lang_name_on_hf}": "dan-Latn", ...},
```
When you transform it into a multilingual dataset, the list should be converted to a mapping.

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        date=None,
+        form=None,
+        domains=None,
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        bibtex_citation=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/dan/WikiClusteringP2P.py:30]: These need to be filled it, the TaskMetadata object contains descriptions of each field.

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/dan/WikiClusteringP2P.py:1]: I would probably either add a script on dataset construction (see the scripts folder) as well as a text description of how the dataset is constructed.

File: docs/mmteb/points.md [hunk 6]
```diff
@@ -27,6 +27,7 @@
 | cassanof          |              | 6           |          |                     | 1          | 1              |           |                |          |               |
 | taidnguyen        |              | 14          |          |                     |            |                |           |                |          |               |
 | orionw         |              |            |          |                     |      2      |              |           |                |          |               |
+| Rysias |              |     24      |          |                    |          |                |         |                |          |              |
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points.md:30]: The points would be:

- 2 for the dataset (reformulated as multilingual)
- 4x3 for new language x task combination (Danish already has a clustering task)

We could probably extend this dataset to more languages, but let us just test it out in this stage first.

You should also add a point for the review (to me / KennethEnevoldsen).

**Review by KennethEnevoldsen — COMMENTED** [review]
Thanks for the addition @Rysias! This looks like a dataset that could promise to cover a wide range of languages. Have added a few comments to work from.

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py (hunk 4 above)
Comment (Rysias) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:30]: In the one I copied, there doesn't seem to be any of these - are there any pointers on how to fill them in for this specific dataset?

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py (hunk 3 above)
Comment (Rysias) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:19]: Great idea! I'll update the pipeline to accommodate:))

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py (hunk 5 above)
Comment (Rysias) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:1]: Good idea, I'll probably link to the repo I've created for generating the dataset and use that as docs:))

File: docs/mmteb/points.md (hunk 6 above)
Comment (Rysias) [reply docs/mmteb/points.md:30]: roger!

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py (hunk 1 above)
Comment (Rysias) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:31]: That might be too little as the 150k comes from 30 splits x 5k sentences. But 500-1k per split might be better?

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py (hunk 4 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:30]: Check out the [adding a dataset](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md) guide or the docstring of the taskmetadata object. Let me know if any of the fields are unclear.

**Comment by KennethEnevoldsen** [comment]
Note for later: This dataset might supersede WikiCitiesClustering

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py (hunk 4 above)
Comment (Rysias) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:30]: Perfect, thank you! I'll fill in all that make sense:))

File: mteb/tasks/Clustering/dan/WikiClusteringP2P.py (hunk 1 above)
Comment (Rysias) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:31]: There are 20-40 categories depending on language. So maybe we could do 10 x 256?

Comment (KennethEnevoldsen) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:31]: I would do it using relatively large splits: 5 x 2048 and then see if reducing it dramatically changes performance.

Comment (Rysias) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py:31]: I have now run it with 30 x 500 and the results are definitely reasonable. I'll try to resample to the suggested and update.

Comment (Rysias) [reply mteb/tasks/Clustering/dan/WikiClusteringP2P.py]: Based on the results, I think I will update to 10 x 1024 and rerun - then I think it's pretty good:))

File: mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        date=None,
+        form=None,
+        domains=["Encyclopaedic"],
+        task_subtypes=None,
+        license="cc-by-sa-3.0",
+        socioeconomic_status="mixed",
+        annotations_creators=None,
+        dialect=None,
+        text_creation="created",
+        bibtex_citation=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py:36]:
```suggestion
        bibtex_citation=None, # None exists
```

File: mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+from mteb.abstasks import AbsTaskClustering, MultilingualTask
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+_LANGUAGES = {
+    "da": ["dan-Latn"],
+    "lv": ["lav-Latn"],
+    "gv": ["glv-Latn"],
+    "sq": ["sqi-Latn"],
+}
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py:11]: Should we expand this?

File: docs/mmteb/points.md [hunk 9]
```diff
@@ -2,6 +2,9 @@
 
 | GitHub            | Total points | New dataset | New task | Dataset annotations | (Bug)fixes | Running Models | Review PR |  Paper Writing | Ideation | Coordination |
 |-------------------| ------------ |-------------| -------- | ------------------- | ---------- | -------------- |-----------| -------------- | -------- | ------------- |
+| KennethEnevoldsen |              | 54          |          |                   8 |         16 |                | 18        |                |          |             5 |
+| x-tabdeveloping   |              | 18          |          |                     |            |                |           |                |          |               |
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points.md:6]: Seems like points have been changed (duplicated) for other contributors. please undo these.

File: mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        dataset={
+            "path": "ryzzlestrizzle/multi-wiki-clustering-p2p",
+            "revision": "7f1d0674aff24ac2b8c3c216e9128ba0f91a5cf4",
+        },
+        type="Clustering",
+        category="p2p",
+        eval_splits=["test"],
+        eval_langs=_LANGUAGES,
+        main_score="v_measure",
+        date=None,
```
[the remark this answers is not in the document]
Comment (Rysias) [reply mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py:27]: Roger, I'll add based on when I created it:))

File: mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py (hunk 8 above)
Comment (Rysias) [reply mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py:11]: We totally could - if you have recommendations for languages I can run them this evening:))

File: docs/mmteb/points.md (hunk 9 above)
Comment (Rysias) [reply docs/mmteb/points.md:6]: Oops, thanks for pointing this out!

File: mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (Rysias) [reply mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py:1]: I have the main docs for this in the repo referenced under `reference` - should I also link to it in the docstring?

**Comment by Rysias** [comment]
Re adding new langs, I'd probably prefer to do this in a separate PR once the main structure and all that has been approved:)). Regarding model runs, I have done MiniLM but am stilling missing the other one - should those results be saved somewhere, @KennethEnevoldsen?

File: mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py:27]: It is data depedent (so it should be the date of the articles). You can probably do start of wiki until now.

File: mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py:1]: Oh that is fine thanks!

File: mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py (hunk 8 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py:11]: https://github.com/embeddings-benchmark/mteb/issues/366

^ anything not covered

**Comment by KennethEnevoldsen** [comment]
> Regarding model runs, I have done MiniLM but am stilling missing the other one - should those results be saved somewhere, @KennethEnevoldsen?

By default it should be in the results folders, you just need to submit them

**Pull request merged**